### DISCUSSION TITLE
CSG.js Alternative - Beginner Needs Help with Basic Setup

### DISCUSSION THREAD

**Discussion by davidh71** [elalish/manifold#1053] [General]
Hey folks,
I'm a beginner/hobby programmer, I'm currently using CSG.js.  I'd like to switch to manifold but as a beginner, I'm having difficulty getting going. Concepts like TypeScript, WebAssembly, and build systems are all new to me.

I've cobbled together the beginnings of a 3d builder using claygl and csg.js - having a **blast** doing it.   I hit a performance wall with csg.js and would love to switch but am struggling to find an example where i can include a javascript file or two and use the power of manifold for some boolean operations.   I have found https://manifoldcad.org and am trying to make sense of the worker & loader & typescript files, but it's over my head.  

Is there any examples I might be missing? 
Appreciate your patience and any help!

Pic of my work-in-progress web project

![Screenshot from 2024-11-17 06-38-16](https://github.com/user-attachments/assets/34ebd3ad-fcde-47bd-84ba-1675dd900d87)

**Comment by elalish** [comment]
I've actually been considering publishing our worker as its own npm package, which would free you from the TS. Basically all you need it call is `await evaluateCADToModel(code: string)`. Does that sound useful? I think it shouldn't be too hard.

**Reply by davidh71** [reply]
Thanks for the response.  Maybe useful?  I'm embarrassed to say i'm not versed in npm, but if that was my only barrier I could figure that out.  Does the worker come with a beginner example - something like https://evanw.github.io/csg.js/more.html  - that was enough for me to figure out how to create a custom mesh, and perform a boolean on it.   
Appreciate your help!

**Reply by elalish** [reply]
The function I mentioned takes verbatim the text in manifoldCAD and turns it into a GLB and 3MF model. So for using the API, just play with manifoldCAD.org.

**Reply by hjhjdev** [reply]
+1 for exporting to package. I'm also having hard time trying to use manifold to CSG boolean from threejs mesh, and then export to 3mf

**Reply by aritzcmz** [reply]
It would be a phenomenal feature for threejs users. Great news! It would be even better if we could use it without the WASM wrapper in a node/deno server environment, but even this would be a huge advance.

**Reply by elalish** [reply]
It is WASM (since Manifold is C++), but it works just fine in Node - that's how we run our JS tests. Anyone interested in helping out with the packaging? I'm a bit short on time at the moment, but I'd be happy to provide guidance.
